### DISCUSSION TITLE
Behavior change request: Selecting an empty line with x.

### DISCUSSION THREAD

**Discussion by demon-venerable** [helix-editor/helix#7890] [General]
Selecting an empty line with x also selects the line under it. This is off-putting and breaks the flow when deleting lines with xd.
I believe the behavior of x only selecting the current line even if it's empty is more intuitive as you don't have to think whether a line is empty or not when deleting it.
For comparision, kakoune's xd and vim's dd only deletes the current line, even if empty.

**Comment by pascalkuthe** [comment]
This was discusses a million times in (closed) issus and PRs right now. Try using the GH search to find some prior discussion of why this command works as it does.

If you don't need the extending behaviour you can use `X` (or even swap the default binding) which doesn't extend to multiple lines and therefore works as you want for empty lines.

**Reply by demon-venerable** [reply]
Thanks for the reply, and sorry for the redundance.

**Reply by CptPotato** [reply]
No worries, here's some of the related issues for context: #835, #3077, #4506
